core/dbt/parser/manifest.py

```diff
@@ -926,6 +930,29 @@ def process_sources(self, current_project: str):
                 continue
             _process_sources_for_exposure(self.manifest, current_project, exposure)
 
+    def process_nodes(self):
+        # make sure the nodes are in the manifest.nodes or the disabled dict,
+        # correctly now that the schema files are also parsed
+        disable_node_copy = deepcopy(self.manifest.nodes)
+        for node in disable_node_copy.values():
```

Comment:
I'm a little concerned by the overhead of deepcopying the entire nodes dictionary. Maybe we could have a list of disabled unique_ids, leave the 'add_disabled_nofile' in place, and just remove the disabled nodes afterward? I'm not so concerned by the disabled dictionary because it will be much smaller.

Reply:
Oh, that's cleaner than creating a list of all nodes.  👍